drivers/gpio/gpio_nrf5.c

```diff
@@ -233,10 +233,10 @@ static int gpio_nrf5_config(struct device *dev,
 		u32_t config = 0;
 		u32_t port = GPIO_PORT(dev);
 
-		if (flags & GPIO_INT_EDGE) {
-			if (flags & GPIO_INT_DOUBLE_EDGE) {
-				config |= GPIOTE_CFG_POL_TOGG;
-			} else if (flags & GPIO_INT_ACTIVE_HIGH) {
+		if (flags & GPIO_INT_DOUBLE_EDGE) {
```

Comment:
The only board using this upstream is:
boards/x86/quark_se_c1000_devboard/board.c

And this board does set GPIO_INT_EDGE alongwith GPIO_INT_DOUBLE_EDGE.

From the include/gpio.h definition, I conclude the intention of the driver model is to classify GPIO_INT_EDGE or GPIO_INT_LEVEL (represented by BIT 5), and GPIO_INT_DOUBLE_EDGE or GPIO_INT_ACTIVE_LOW or GPIO_INT_ACTIVE_HIGH (represented by a 2-bit value, BIT 6 and 2).

Reply:
Okay, so we interpret the documentation differently.

Perhaps we can ask the author ...

Reply:
Hi,

this PR is assuming that DOUBLE_EDGE does not require EDGE and updating gpio_nrf5.c accordingly. If there are users that have assumed otherwise then they will have to be updated as well, but IMHO that should be done independently of this PR.

I could create issues for the users we know of that have interpreted the API differently though.